Documentation/cassandra-operator-upgrade.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Cassandra Operator Upgrade
+weight: 51
+---
+
+# Cassandra Operator Upgrades
+
+This guide will walk you through the manual steps to upgrade the software in Cassandra Operator from one version to the next. The cassandra operator is made up of two parts: 
+1. The main controller that runs as a standalone application, watches the Cassandra Cluster CRD and makes administrative decisions. We'll call this component `Controller` for the rest of this document.
+2. A sidecar that runs alongside each member of a Cassandra Cluster. We will call this component `Sidecar`.
+
+Both components should be updated. This is a very manual process at the moment, but it should be automated soon in the future, once the Cassandra Operator reaches the beta stage.
+
+## Considerations
+With this upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is not without risk. There may be unexpected issues or
+  obstacles that damage the integrity and health of your storage cluster, including data loss. Only
+  proceed with this guide if you are comfortable with that. It is recommended that you backup your data before proceeding.
+* **WARNING:** The current process to upgrade REQUIRES the cluster to be unavailable for the time of the upgrade.
+
+## Prerequisites
+
+* Before starting the procedure, ensure that your Cassandra Clusters are in a healthy state. You can check a Cassandra Cluster's health by using `kubectl describe $NAME -n $NAMESPACE` and ensuring that for each rack in the Status, `readyMembers` equals `members`.
+* If you are upgrading from v0.9.1 to a later version, the mount point of the PVC for each member has changed because it was wrong. 
+
+## Procedure
+
+1. Because each version of the `Controller` is designed to work with the same version of the `Sidecar`, they must be upgraded together. In order to avoid mixing versions between the `Controller` and `Sidecar`, we first delete every Cassandra Cluster CRD in our Kubernetes cluster, after first backing up their manifests. This will not delete your data because the PVCs will be retained even if the Cassandra Cluster object is deleted. Example:
+```bash
+# Assumes cluster rook-cassandra in namespace rook-cassandra
+NAME=rook-cassandra
+NAMESPACE=rook-cassandra
+
+kubectl get $NAME -n $NAMESPACE -o yaml > $NAME.yaml
+kubectl delete $NAME -n $NAMESPACE
+```
+
+2. After that, we upgrade the version of the `Controller`. To achieve that, we patch the StatefulSet running the `Controller`:
```

Comment:
"Controller"? It is about the Rook Cassandra Operator, right? Just writing operator or the full would be less confusing (same for the first list point).

Reply:
This is a naming thing (always the worst :stuck_out_tongue: )
The pattern of using a controller and a CRD is called the Operator pattern. 
The software running, looping and watching the CRD is called a Controller. 
But since operator is heavily used in the rook docs, maybe it's better to call it that.
@jbw976 what do you think?

Reply:
in my experience (within the Rook community especially), the common understanding is that Operator is used to cover all of that, both the pattern and the controller itself.  I would use the terminology of "operator" in this document here to avoid confusion.